compiler/il/OMRNode.cpp

```diff
@@ -1192,6 +1208,38 @@ OMR::Node::createArraycopy(TR::Node *first, TR::Node *second, TR::Node * third,
    }
 
 
+TR::Node *
+OMR::Node::createPotentialOSRPointHelperCallInILGen(TR::Node* originatingByteCodeNode, int32_t offset)
+   {
+   TR::Compilation* comp = TR::comp();
+   // The following are assertions to prevent misuses of this helper
+   //
+   TR_ASSERT(comp->getCurrentIlGenerator(), "This API must be called during ILGen");
+   TR_ASSERT(!comp->isPeekingMethod(), "Can not generate the helper call during peeking");
+   TR_ASSERT(comp->supportsInduceOSR(), "Can not create the helper without OSR support");
+
+   TR::Node* offsetNode = TR::Node::iconst(originatingByteCodeNode, offset);
+   TR::Node* callNode = TR::Node::createWithSymRef(originatingByteCodeNode, TR::call, 1, offsetNode, TR::comp()->getSymRefTab()->findOrCreatePotentialOSRPointHelperSymbolRef());
+
+   // Node created outside of ilgen will have doNotProfile set, this results in OSR infrastructure believing it
+   // can not OSR at this node. Reset this flag since a potentialOSRPointHelper is a safe OSR transition point
+   //
+   callNode->getByteCodeInfo().setDoNotProfile(0);
```

Comment:
Now that this function is restricted to ilgen, does it still need to `setDoNotProfile(0)`?

Reply:
The Node constructor sets `doNotProfile` if an `originatingByteCodeNode` is passed in.